### PR TITLE
Invoke Karma directly and add CI-mode npm script

### PR DESCRIPTION
**Pull request by kadamwhite**
Using `grunt-karma` as an intermediary does not gain us anything in terms of performance, and actually complicates the configuration because several parameters must be explicitly set in both the karma config _and_ grunt config in order for the tests to run as expected.

This PR removes grunt-karma and aliases `npm test` to karma directly.

It also adds an `npm run ci` command, which runs the tests in Firefox (PhantomJS is unsuitable due to the dependence on modern DOM features) in single-run ('continuous integration') mode.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "test": "tests"
   },
   "scripts": {
-    "test": "grunt test-dev"
+    "ci": "karma start karma.conf.js --single-run",
+    "test": "karma start karma.conf.js"
```
Comment (mzgoddard) [inline package.json]: I think test should be our single run. Maybe that is an alignment with travis and other CI tools that default to npm test but I think that also colors what `npm test` means to me. A single run through the tests.

Comment (mzgoddard) [inline package.json]: `npm test-dev` is what I use for non-single-runs in a lot of my projects. I'm also super open to any other names.

Comment (kadamwhite) [inline package.json]: I've used both `test-dev` and `test:dev` before, no strong preference.

Comment (mzgoddard) [inline package.json]: I'm happy with either of those you want to pick.

**Comment by mzgoddard** [comment]
👍

**Pull request merged**